hw/xbox/xid-sdl.c

```diff
@@ -194,21 +194,22 @@ static const XIDDesc desc_xid_xbox_gamepad = {
 
 #define BUTTON_MASK(button) (1 << ((button) - GAMEPAD_DPAD_UP))
 
+SDL_HapticEffect effect;
+
 static void update_output(USBXIDState *s) {
 #ifdef FORCE_FEEDBACK
-    if (s->sdl_haptic == NULL) { return; }
-
-    SDL_HapticLeftRight effect = {
-        .type = SDL_HAPTIC_LEFTRIGHT,
-        .length = SDL_HAPTIC_INFINITY,
-        /* FIXME: Might be left/right inverted */
-        .large_magnitude = s->out_state.right_actuator_strength,
-        .small_magnitude = s->out_state.left_actuator_strength
-    };
+    if (s->sdl_haptic == NULL) {
+        return;
+    }
+
+    SDL_memset( &effect, 0, sizeof(SDL_HapticEffect) ); // 0 is safe default
```

Comment:
For whatever reason, passing in `SDL_HapticLeftRight` to `SDL_HapticNewEffect` failed, setting `effect_id` to `-1`. Using `SDL_HapticEffect` with `leftright` params seems to work fine instead.

We probably don't need to allocate it this way though. I was comparing the control flow to [this implementation.](https://github.com/JayFoxRox/sdl2-jstest-haptic) I was being too impatient testing the rumble within xqemu with the game I was testing with. It took a good 20-30 seconds after the game booted for `SDL_HapticRunEffect` to rumble my controller even with hard coded `large_magnitude`. The solution here isn't necessary but `sdl2-jstest-haptic` worked so I wanted to get as close to that as possible.

Reply:
Please check if we can make it work with the smaller struct, or how this is intended in SDL (by checking their examples or contacting the API author [should be @bobbens])